config/v1/types_dns.go

```diff
@@ -96,3 +103,44 @@ type DNSList struct {
 
 	Items []DNS `json:"items"`
 }
+
+// DNSPlatformType is a cloud infrastructure provider that supports
+// platform specific DNS configuration.
+// +kubebuilder:validation:Enum=AWS
+type DNSPlatformType PlatformType
```

Comment:
Is this ok? Or would you prefer
```suggestion
type DNSPlatformType string
```
I was hoping this would allow 

```go
	AWSDNSPlatformType DNSPlatformType = AWSPlatformType
```
but that does not seem to work

Reply:
PlatformType is fine.  I think you can add regex validation directly on your field.  If not, CEL would work ok in this instance.